packages/flutter_tools/lib/src/hot.dart

```diff
@@ -467,25 +459,41 @@ class HotRunner extends ResidentRunner {
   }
 
   /// Returns [true] if the reload was successful.
-  bool _printReloadReport(Map<String, dynamic> reloadReport) {
+  bool _validateReloadReport(Map<String, dynamic> reloadReport) {
     if (!reloadReport['success']) {
       printError('Hot reload was rejected:');
       for (Map<String, dynamic> notice in reloadReport['details']['notices'])
         printError('${notice['message']}');
       return false;
     }
-    int loadedLibraryCount = reloadReport['details']['loadedLibraryCount'];
-    int finalLibraryCount = reloadReport['details']['finalLibraryCount'];
-    printStatus('Reloaded $loadedLibraryCount of $finalLibraryCount libraries.');
     return true;
   }
 
   @override
   Future<OperationResult> restart({ bool fullRestart: false, bool pauseAfterRestart: false }) async {
     if (fullRestart) {
-      return _restartFromSources();
+      Status status = logger.startProgress('Performing full restart...');
+      try {
+        await _restartFromSources();
+        status.stop();
+        printStatus('Restart complete.');
+        return OperationResult.ok;
+      } catch (error) {
+        status.stop();
+        throw error;
+      }
     } else {
-      return _reloadSources(pause: pauseAfterRestart);
+      Status status = logger.startProgress('Performing hot reload...');
+      try {
+        OperationResult result = await _reloadSources(pause: pauseAfterRestart);
+        status.stop();
+        if (result.isOk)
+          printStatus("${result.message}.");
+        return result;
+      } catch (error) {
```

Comment:
Makes sense. I updated this to use `rethrow`; I'm sticking with catch instead of finally since I want a little more control of when stop() is called.